features/logging/4.x/collector.feature

```diff
@@ -0,0 +1,62 @@
+@clusterlogging
+Feature: collector related tests
+
+  # @auther qitang@redhat.com
+  # case_id OCP-25365
+  @admin @destructive
+  @commonlogging
+  Scenario: The System Journald log can be collected
+    Given I switch to cluster admin pseudo user
+    Given I use the "openshift-logging" project
+    Given evaluation of `cluster_logging('instance').collection_type` is stored in the :collection_type clipboard
+    And I wait up to 600 seconds for the steps to pass:
+    """
+    And I get the ".operations" logging index information from a pod with labels "es-node-master=true"
+    Then the step should succeed
+    And the expression should be true> cb.index_data and cb.index_data.count > 0
+    """
+    When I perform the HTTP request on the ES pod with labels "es-node-master=true":
+      | relative_url | .operation*/_search?pretty'  -d '{"query": {"exists": {"field": "systemd"}}} |
+      | op           | GET                                                                          |
+    Then the step should succeed
+    And the expression should be true> @result[:parsed]['hits']['hits'][0]['_source']['pipeline_metadata']['collector']['name'] == cb.collection_type
+    And the expression should be true> @result[:parsed]['hits']['hits'][0]['_source']['pipeline_metadata']['collector']['inputname'] == (cb.collection_type == "fluentd" ? "fluent-plugin-systemd" : "imfile")
+
+  # @auther qitang@redhat.com
+  # @case_id OCP-18147
+  @admin @destructive
+  @commonlogging
+  Scenario: The Container logs metadata check
+    Given I switch to the first user
+    Given I create a project with non-leading digit name
+    Given evaluation of `project` is stored in the :proj clipboard
+    When I run the :new_app client command with:
+      | file | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/loggen/container_json_unicode_log_template.json |
+    Then the step should succeed
+    And I wait up to 600 seconds for the steps to pass:
+    """
+    And a pod becomes ready with labels:
+      | run=centos-logtest,test=centos-logtest |
+    """
+    Given evaluation of `pod` is stored in the :log_pod clipboard
+    Given I switch to cluster admin pseudo user
+    Given I use the "openshift-logging" project
+    Given evaluation of `cluster_logging('instance').collection_type` is stored in the :collection_type clipboard
+    And I wait up to 600 seconds for the steps to pass:
+    """
+    And I get the "project.<%= cb.proj.name %>" logging index information from a pod with labels "es-node-master=true"
+    Then the step should succeed
+    And the expression should be true> cb.index_data and cb.index_data.count > 0
+    """
```

Comment:
@pruan-rht , I see this check two times already. As if it waits for logging data to be present. Do we have a special step?

@QiaolingTang , cold you confirm purpose of this waiting?

Reply:
@akostadinov the ES may need some time to create the index and sometimes the collector needs time to collect the logs in projects, these condition always happen in a shared cluster due to limited resources, so I added the step to wait for the data to appear in the ES.

Reply:
@QiaolingTang to be consistent, let's assume we are always do this check so it's better to wrap it in a step as suggested by @akostadinov

Reply:
@pruan-rht , so step like `Given log entries appear for the project`?

Reply:
@pruan-rht @akostadinov seems we have a step `When /^I wait(?: (\d+) seconds)? for the #{QUOTED} index to appear in the ES pod(?: with labels #{QUOTED})?$/ do |seconds, index_name, pod_labels|`, I'll try this step.

Reply:
Awesome, good catch!

Reply:
@QiaolingTang yeah, that was written a while back.